src/batch.js

```diff
@@ -1,28 +1,221 @@
 'use strict'
 
+const router = require('lambda-router/src/router')
+const { expectation } = require('sinon')
+
 const { HttpError } = require('./httpError')
 
 const batchDefaultConfig = {
-  maxBatchSize: 20
+  maxBatchSize: 20,
+  handler: undefined
 }
 
 module.exports = {
   batchHandler
 }
 
-async function batchHandler({ routes, config, extractPathParameters }, event, context) {
+async function batchHandler({ route, config }, event, context) {
```

Comment:
routes was a collection of all the routes and handlers, however figuring out if a url (_/user/id1_) matches the routes urls (_/user/{id}_) would require duplicating most of the logic from the router.js. Instead I opted to just provide the router function and call it to share the logic.